### ISSUE TITLE
[Question] Enabling 'init_model_with_meta_device' results in random weights despite 'load_weights=True'

### ISSUE THREAD

**Issue by v-dicicco** [NVIDIA-NeMo/Megatron-Bridge#1552]
Hi!
I'm testing finetuning with FSDP on multiple 4xH100 (64GB) nodes. I'm using a configuration similar to the following (loaded from a yaml but then merged in a `ConfigContainer`)

```yaml
model:
  [...]
  seq_length: 2048
  tensor_model_parallel_size: 1
  context_parallel_size: 1
  sequence_parallel: false
  pipeline_model_parallel_size: 1

optimizer:
  [...]
  use_distributed_optimizer: true

ddp:
  [...]
  use_distributed_optimizer: true
  average_in_collective: false
  use_megatron_fsdp: true
  data_parallel_sharding_strategy: "optim_grads_params"

checkpoint:
  [...]
  ckpt_format: "fsdp_dtensor"   # required by FSDP
```

I've tested the configuration with Llama3.1 8B and it works as expected, loading the model & weights from an hf checkpoint with:
```python
bridge = AutoBridge.from_hf_pretrained(hf_path)
model_cfg = bridge.to_megatron_provider(load_weights=True)
```

However, as soon as I switch to Llama3.1 70B I get an OOM during model setup, __even before the training or the dataset preprocessing starts__.

I've tried `use_cpu_initialization` without luck, and the only way I found to avoid this OOM is by enabling `init_model_with_meta_device` taking inspiration from [this FSDP  configuration](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/da980700aaef8de155155b00ed56fd78b1677b83/scripts/performance/utils/helpers.py#L116-L124):

```yaml
model:
  [...]
  init_model_with_meta_device: true
```

Although the training starts and continues correctly, I'm observing a high loss that clearly indicates that the original weights are not loaded in the model anymore (I can reproduce the same with the 8B model and as soon as i enable the meta device init the loss increases).

Same happens if i set:
```yaml
checkpoint:
  [...]
  pretrained_checkpoint: /path/to/a/converted/megatron/checkpoint/iter_0000000/
```

What is the recommended way to init a "big" model with FSDP to avoid OOM while keeping the original weights?

**Comment by BoxiangW** [comment]
Hi, thanks for the issue, I am trying to reproduce and will get back to you asap. It would be also great if you can share a repro script
